### PR TITLE
MON-5286 broker stats working on remote server

### PR DESCRIPTION
**Pull request by garnier-quentin**
## Description

https://github.com/centreon/centreon/issues/8488

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [X] 19.10.x
- [X] 20.04.x (master)

## Checklist

- [X] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/config-generate-remote/Broker.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
+<?php
```
Comment (sc979) [inline www/class/config-generate-remote/Broker.php:1]:
```suggestion
<?php

```

File: www/class/config-generate-remote/Broker.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ */
+
+namespace ConfigGenerateRemote;
+
+use \Exception;
+use \PDO;
+use ConfigGenerateRemote\Abstracts\AbstractObject;
+
+class Broker extends AbstractObject
```
Comment (sc979) [inline www/class/config-generate-remote/Broker.php:27]: PSR12 may need a docblock. But I'm not sure about it

File: www/class/config-generate-remote/Relations/BrokerInfo.php [hunk 3]
```diff
@@ -0,0 +1,147 @@
+<?php
+/*
```
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:2]:
```suggestion

/*
```

File: www/class/config-generate-remote/Relations/BrokerInfo.php [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+ * For more information : contact@centreon.com
+ *
+ */
+
+namespace ConfigGenerateRemote\Relations;
+
+use \PDO;
+use ConfigGenerateRemote\Abstracts\AbstractObject;
+
+class BrokerInfo extends AbstractObject
```
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:26]: same coment

File: www/class/config-generate-remote/Relations/BrokerInfo.php [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    }
+
+    /**
+     * Build cache
+     *
+     * @return void
+     */
+    private function buildCache()
+    {
+        if ($this->doneCache == 1) {
```
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:89]: why not use true/false ?

Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:89]:
```suggestion
        if ($this->doneCache === 1) {
```

File: www/class/config-generate-remote/Relations/BrokerInfo.php [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+     * @param  integer $configId
+     * @return array
+     */
+    public function getBrokerInfoByConfigId(int $configId)
+    {
+        # Get from the cache
+        if (isset($this->brokerInfoCache[$configId])) {
+            $this->generateObject($configId, $this->brokerInfoCache);
+            return $this->brokerInfoCache[$configId];
+        } elseif ($this->useCache == 1) {
```
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:123]:
```suggestion
        } elseif ($this->useCache === 1) {
```

File: www/class/config-generate-remote/Relations/BrokerInfo.php [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+            return null;
```
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:124]: shouldn't you return an empty array instead ? (docblock description)

File: www/class/config-generate-remote/Relations/BrokerInfo.php [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+    /**
+     * Get broker information config
+     *
+     * @param  integer $configId
+     * @return array
+     */
+    public function getBrokerInfoByConfigId(int $configId)
+    {
+        # Get from the cache
```
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:119]:
```suggestion
        // Get from the cache
```

File: www/class/config-generate-remote/Relations/BrokerInfo.php [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+        if (isset($this->brokerInfoCache[$configId])) {
+            $this->generateObject($configId, $this->brokerInfoCache);
+            return $this->brokerInfoCache[$configId];
+        } elseif ($this->useCache == 1) {
+            return null;
+        }
+
+        # We get unitary
```
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:127]:
```suggestion
        // We get unitary
```

Comment (sc979) [reply www/class/config-generate-remote/Relations/BrokerInfo.php:127]: what does this mean ?

File: www/class/config-generate-remote/Relations/BrokerInfo.php (hunk 5 above)
Comment (garnier-quentin) [reply www/class/config-generate-remote/Relations/BrokerInfo.php]: same usage in other class. be consistent

File: www/class/config-generate-remote/Relations/BrokerInfo.php [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+    /**
+     * Build cache
+     *
+     * @return void
+     */
+    private function buildCache()
+    {
+        if ($this->doneCache === 1) {
+            return 0;
```
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:91]: why return 0 when you don't use it and document @return void ?

File: www/class/config-generate-remote/Relations/BrokerInfo.php [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+        }
+
+        $this->cacheBrokerInfo();
+        $this->doneCache = 1;
```
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:95]:
```suggestion
        if ($this->doneCache !== 1) {
            $this->cacheBrokerInfo();
            $this->doneCache = 1;
        }
```

File: www/class/config-generate-remote/Relations/BrokerInfo.php (hunk 10 above)
Comment (sc979) [inline www/class/config-generate-remote/Relations/BrokerInfo.php:91]: why return 0 when you don't use it and document @return void ?

**Pull request merged**